### PR TITLE
feat: Add implementation for Sponsor updates #934

### LINKED ISSUE #934: Allow updates to sponsor details
**Is it a ?**

- [ ] Bug Report

- [x] Feature Request
    
- [ ] Chore
    
### Bug Report    

**Feature Description**

The API supports update operation and the app needs to allow the organizer to update Sponsor details.

**Parent issue**
#730 

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by mishuvs**
Fixes #934 

Changes: Allows updates to sponsors

Screenshots for the change: 

videotogif_2018 05 13_17 20 11

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/app/core/sponsor/update/UpdateSponsorFragment.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+import javax.inject.Inject;
+
+import br.com.ilhasoft.support.validation.Validator;
+import dagger.Lazy;
+
+import static org.fossasia.openevent.app.ui.ViewUtils.showView;
+
+public class UpdateSponsorFragment extends BaseBottomSheetFragment<UpdateSponsorPresenter> implements UpdateSponsorView {
+
+    private static final String TRACK_ID = "id";
```
Comment (sridharjajoo) [inline app/src/main/java/org/fossasia/openevent/app/core/sponsor/update/UpdateSponsorFragment.java]: I think this should be SPONSOR_ID?

Comment (mishuvs) [reply app/src/main/java/org/fossasia/openevent/app/core/sponsor/update/UpdateSponsorFragment.java]: Yes, btw I had specified "WIP"

**Comment by mishuvs** [comment]
@iamareebjamal Please review.

**Comment by iamareebjamal** [comment]
Should update automatically without needing to refresh

**Comment by mishuvs** [comment]
@iamareebjamal I have updated the PR and the gif. Please review.

**Comment by iamareebjamal** [comment]
This should not trigger the reload. Instead, change listener should update the view itself

**Comment by mishuvs** [comment]
@iamareebjamal I think it is listening for changes and after the changeListener notifies, loadSponsors() is called. This is also consistent with how the other presenters handle changes.

**Comment by mishuvs** [comment]
@iamareebjamal Please merge this PR so that I can send PRs for deletion and tests without conflicts.

**Pull request merged**